pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -169,42 +183,37 @@ func (h *RegistrationHandler) validateVersions(
 	if existingPlugin.highestSupportedVersion.LessThan(newPluginHighestVersion) {
 		return newPluginHighestVersion, nil
 	}
-	return nil, errors.New(
-		log(
-			"%s for DRA plugin %q failed. Another plugin with the same name is already registered with a higher supported version: %q",
-			callerName,
-			pluginName,
-			existingPlugin.highestSupportedVersion,
-		),
-	)
-}
-
-func deregisterPlugin(pluginName string) {
-	draPlugins.delete(pluginName)
+	return nil, fmt.Errorf("another plugin instance is already registered with a higher supported version: %q < %q", newPluginHighestVersion, existingPlugin.highestSupportedVersion)
 }
 
 // DeRegisterPlugin is called when a plugin has removed its socket,
 // signaling it is no longer available.
 func (h *RegistrationHandler) DeRegisterPlugin(pluginName string) {
-	klog.InfoS("DeRegister DRA plugin", "name", pluginName)
-	deregisterPlugin(pluginName)
-	h.controller.removePlugin(pluginName)
+	if p := draPlugins.delete(pluginName); p != nil {
+		logger := klog.FromContext(p.backgroundCtx)
+		logger.V(3).Info("Deregister DRA plugin", "endpoint", p.endpoint)
+
+		// Did exist before. Let's clean up ResourceSlices for it because
+		// it might have died without doing that itself and might not
+		// come back.
```

Comment:
Can we change this to:
> Clean up the ResourceSlices for the deleted Plugin since it may have died without doing so itself and might never come back.

Reply:
Done.